src/graphnet/models/model.py

```diff
@@ -79,15 +81,32 @@ def fit(
         *,
         max_epochs: int = 10,
         gpus: Optional[Union[List[int], int]] = None,
-        callbacks: Optional[List[Callback]] = None,
+        callbacks: Optional[List[Callback]] = [ProgressBar()],
         ckpt_path: Optional[str] = None,
         logger: Optional[LightningLogger] = None,
         log_every_n_steps: int = 1,
         gradient_clip_val: Optional[float] = None,
         distribution_strategy: Optional[str] = "ddp",
+        early_stopping_patience: int = 5,
         **trainer_kwargs: Any,
     ) -> None:
         """Fit `Model` using `pytorch_lightning.Trainer`."""
+        # Checks
+        if val_dataloader is not None:
+            has_es = False
+            assert isinstance(callbacks, list)
+            for callback in callbacks:
+                if isinstance(callback, EarlyStopping):
+                    has_es = True
+            if has_es is False:
+                callbacks.append(
+                    EarlyStopping(
+                        monitor="val_loss",
+                        patience=early_stopping_patience,
+                    )
+                )
+                self.info("EarlyStopping callback added automatically.")
+
```

Comment:
I get the point of adding the `ProgressBar` by default, as it has no functional impact. However, I'm a little bit unsure about this one: So far, we have chosen to set callback based on the `callbacks` argument, whereby the user needs to specify the list of callbacks they want, and is quite common practice. Now, we're considering mixing this pattern with adding callbacks based on specific, additional arguments (here: `early_stopping_patience`). I think we should do one or the other, i.e., setting callbacks through a user-provided list _or_ through dedicated arguments to `Model.fit`.